akka-stream-tests/src/test/scala/akka/stream/scaladsl/SubstreamSubscriptionTimeoutSpec.scala

```diff
@@ -74,7 +73,8 @@ class SubstreamSubscriptionTimeoutSpec(conf: String) extends AkkaSpec(conf) {
       // sleep long enough for it to be cleaned up
       Thread.sleep(1500)
 
-      val f = s3.runWith(Sink.head).recover { case _: SubscriptionTimeoutException ⇒ "expected" }
+      // Must be a Sink.seq, otherwise there is a race due to the concat in the `lift` implementation
+      val f = s3.runWith(Sink.seq).recover { case _: SubscriptionTimeoutException ⇒ "expected" }
```

Comment:
Yes, but do you know a better way to test this? Currently we rely on that the 300ms subscription timeout does not overcome the 1500 ms sleep.